api/src/permissions/lib/fetch-policies.test.ts

```diff
@@ -0,0 +1,107 @@
+import type { Accountability } from '@directus/types';
+import { beforeEach, expect, test, vi } from 'vitest';
+import { AccessService } from '../../services/access.js';
+import type { AccessRow } from '../modules/process-ast/types.js';
+import type { Context } from '../types.js';
+import { _fetchPolicies as fetchPolicies } from './fetch-policies.js';
+
+vi.mock('../../services/access.js', () => ({
+	AccessService: vi.fn(),
+}));
+
+let rows: AccessRow[];
+
+beforeEach(() => {
+	rows = [];
+
+	AccessService.prototype.readByQuery = vi.fn().mockResolvedValue(rows);
+});
+
```

Comment:
Update test for changes in https://github.com/directus/directus/pull/22773/files#r1647286436

```suggestion

test('Fetches policies for public role and user when user is given without role', async () => {
	const acc = { roles: [], user: 'user-a' } as unknown as Accountability;

	const policies = await fetchPolicies(acc, {} as Context);

	expect(AccessService.prototype.readByQuery).toHaveBeenCalledWith({
		filter: {
			_or: [
				{
					role: {
						_null: true,
					},
					user: {
						_null: true,
					},
				},
				{ user: { _eq: 'user-a' } },
			],
		},
		fields: ['policy.id', 'policy.ip_access'],
		limit: -1,
	});

	expect(policies).toEqual([]);
});

```